src/ds/d500/d500-mapping.cpp

```diff
@@ -0,0 +1,252 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include "d500-mapping.h"
+
+#include <mutex>
+#include <chrono>
+#include <vector>
+#include <map>
+#include <iterator>
+#include <cstddef>
+
+#include "ds/ds-timestamp.h"
+#include "ds/ds-options.h"
+#include "stream.h"
+
+namespace librealsense
+{
+    
+    const std::map<uint32_t, rs2_format> mapping_fourcc_to_rs2_format = {
+        {rs_fourcc('G','R','E','Y'), RS2_FORMAT_OCCUP},
+        //point cloud - needed a standard format, with 32 bit per pixel:
+        // #define V4L2_PIX_FMT_ARGB32  v4l2_fourcc('B', 'A', '2', '4') /* 32  ARGB-8-8-8-8  */
+        {rs_fourcc('B','A','2','4'), RS2_FORMAT_XYZ32F}
+    };
+    const std::map<uint32_t, rs2_stream> mapping_fourcc_to_rs2_stream = {
+        {rs_fourcc('G','R','E','Y'), RS2_STREAM_OCCUPANCY},
+        {rs_fourcc('B','A','2','4'), RS2_STREAM_POINT_CLOUD}
+    };
+
+    d500_mapping::d500_mapping(std::shared_ptr<context> ctx,
+        const platform::backend_device_group& group)
+        : device(ctx, group), d500_device(ctx, group),
+        _occupancy_stream(new stream(RS2_STREAM_OCCUPANCY)),
+        _point_cloud_stream(new stream(RS2_STREAM_POINT_CLOUD))
+    {
+        using namespace ds;
+
+        const uint32_t occupancy_stream_mi = 13;
+        auto occupancy_devs_info = filter_by_mi(group.uvc_devices, occupancy_stream_mi);
+        
+        if (occupancy_devs_info.size() != 1)
+            throw invalid_value_exception(rsutils::string::from() << "RS5XX with Safety models are expected to include a single occupancy device! - "
```

Comment:
Safety model? maybe module?

Reply:
This comment has been copied from other sensors. The meaning is models - דגמים. 
I  ll improve it